tests/testthat/test-linear_pool.R

```diff
@@ -411,6 +426,195 @@ test_that("(weighted) quantiles correctly calculated - lognormal family", {
+    dplyr::mutate(horizon = 0, value = 0.75 * value) |>
+    dplyr::filter(model_id %in% letters[1:3]) |>
+    dplyr::bind_rows(sample_outputs)
+
+  sample_outputs |>
+    linear_pool(
+      weights = NULL,
+      task_id_cols = c("target_date", "target", "horizon", "location"),
+      compound_taskid_set = c("target", "location", "target_date"),
+      n_output_samples = NULL
+    ) |>
+    expect_error(
+      regex = "Not all component models in `model_out_tbl` forecast for the same set of dependent tasks",
+      fixed = TRUE
+    )
+
+  # test that df of missing combos returns the expected value
+  missing_expected <- data.frame(model_id = letters[4], horizon = 0)
+  missing_actual <- sample_outputs |>
+    validate_compound_taskid_set(
+      task_id_cols = c("target_date", "target", "horizon", "location"),
+      compound_taskid_set = c("target", "location", "target_date"),
+      return_missing_combos = TRUE
+    )
+  expect_equal(missing_actual, dplyr::tibble(missing_expected))
+})
+
+
+test_that("If the specified `compound_taskid_set` is incompatible with component model outputs, throw an error", {
+  sample_outputs <- expand.grid(stringsAsFactors = FALSE,
+                                model_id = letters[1:4],
+                                location = c("222", "888"),
+                                horizon = 1, #week
+                                target = "inc death",
+                                target_date = as.Date("2021-12-25"),
+                                output_type = "sample",
+                                output_type_id = 1:3,
+                                value = NA_real_)
+
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 1] <-
+    c(40, 30, 45, 80)
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 2] <-
+    c(60, 40, 75, 20)
+  sample_outputs$value[sample_outputs$location == "222" &
+                         sample_outputs$output_type_id == 3] <-
+    c(10, 70, 15, 50)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 1] <-
+    c(100, 325, 400, 300)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 2] <-
+    c(250, 350, 500, 250)
+  sample_outputs$value[sample_outputs$location == "888" &
+                         sample_outputs$output_type_id == 3] <-
+    c(150, 300, 500, 350)
+
+  sample_outputs |>
+    dplyr::mutate(horizon = 0, value = 0.75 * value) |>
+    dplyr::bind_rows(sample_outputs) |>
+    dplyr::mutate(output_type_id = paste0(.data[["location"]], .data[["output_type_id"]])) |>
+    linear_pool(
+      weights = NULL,
+      task_id_cols = c("target_date", "target", "horizon", "location"),
+      compound_taskid_set = c("target", "target_date"),
+      n_output_samples = NULL
+    ) |>
+    expect_error(
```

Comment:
I made a comment in the next test about piping an error-generating function call into `expect_error` that also applies here.